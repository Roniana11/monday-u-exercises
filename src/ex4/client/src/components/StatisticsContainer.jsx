import { Flex } from 'monday-ui-react-core';
import 'monday-ui-react-core/dist/main.css';
import { useState, useEffect } from 'react';
import { getItems } from '../Services/item_client';
import LoadingSpinner from './ui/LoadingSpinner';

const StatisticsContainer = () => {
  const [items, setItems] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const list = [];
    getItems().then((result) => {
      result.forEach((item) => list.push(item));
      setIsLoading(false);
      setItems(list);
    });
  }, []);

  return (
    <Flex
      className="todo-container"
      justify={Flex.justify.SPACE_BETWEEN}
      direction={Flex.directions.COLUMN}
    >
      <div className="todo-list-container">
        <h1>Statistics</h1>
        {isLoading && <LoadingSpinner />}
        {!isLoading && (
          <div>
            <h3>Completed: </h3>
            <h3>{items.filter((item) => item.status).length}</h3>
            <h3>Overall:</h3>
            <h3> {items.length}</h3>
            <h3>Catched pokemons:</h3>
            <h3>
              {items.filter((item) => item.pokemonId && item.status).length}
            </h3>
          </div>
        )}
      </div>
    </Flex>
  );
};

export default StatisticsContainer;
