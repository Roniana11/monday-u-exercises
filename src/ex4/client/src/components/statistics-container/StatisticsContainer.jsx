import { Flex } from 'monday-ui-react-core';
import 'monday-ui-react-core/dist/main.css';
import { useSelector } from 'react-redux';

const StatisticsContainer = () => {
  const items = useSelector((state) => state.itemsEntities.items);

  return (
    <Flex
      className="todo-container"
      justify={Flex.justify.SPACE_BETWEEN}
      direction={Flex.directions.COLUMN}
    >
      <div className="todo-list-container">
        <h1>Statistics</h1>
        <div>
          <h3>Completed: </h3>
          <h3>{items.filter((item) => item.status).length}</h3>
          <h3>Overall:</h3>
          <h3> {items.length}</h3>
          <h3>Catched pokemons:</h3>
          <h3>
            {items.filter((item) => item.pokemonId && item.status).length}
          </h3>
        </div>
      </div>
    </Flex>
  );
};

export default StatisticsContainer;
