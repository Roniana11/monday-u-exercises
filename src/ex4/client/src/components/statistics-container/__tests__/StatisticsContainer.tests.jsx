import renderer from 'react-test-renderer';
import StatisticsContainer from '../StatisticsContainer';
import { Provider } from 'react-redux';
import { store } from '../../../store';

test('renders correctly the StatisticsContainer', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <StatisticsContainer />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
