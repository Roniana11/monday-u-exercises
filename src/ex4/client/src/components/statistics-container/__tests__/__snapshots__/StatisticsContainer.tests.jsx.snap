// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the StatisticsContainer 1`] = `
<div
  className="components-Flex-Flex-module__container--ZDN6j components-Flex-Flex-module__directionColumn--VHI6B components-Flex-Flex-module__justifySpaceBetween--lQ3bb components-Flex-Flex-module__alignCenter--oWj46 todo-container"
  style={
    Object {
      "gap": "0px",
    }
  }
>
  <div
    className="todo-list-container"
  >
    <h1>
      Statistics
    </h1>
    <div>
      <h3>
        Completed: 
      </h3>
      <h3>
        0
      </h3>
      <h3>
        Overall:
      </h3>
      <h3>
         
        0
      </h3>
      <h3>
        Catched pokemons:
      </h3>
      <h3>
        0
      </h3>
    </div>
  </div>
</div>
`;
